pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -46,7 +46,7 @@ func ValidateMachinePool(p *openstack.MachinePool, ci *CloudInfo, controlPlane b
 		}
 
 		allErrs = append(allErrs, validateZones(p.RootVolume.Zones, ci.VolumeZones, fldPath.Child("rootVolume").Child("zones"))...)
-		if len(p.RootVolume.Zones) != 1 && len(p.RootVolume.Zones) != len(p.Zones) {
+		if len(p.RootVolume.Zones) > 1 && len(p.RootVolume.Zones) != len(p.Zones) {
```

Comment:
Just trying to work out the expected semantics of p.RootVolume.Zones here. Below in validateZones we have:

``` go
func validateZones(input []string, available []string, fldPath *field.Path) field.ErrorList {
        // check if machinepool default
        if len(input) == 1 && input[0] == "" {
                return field.ErrorList{}
        }
...
```

which suggests that 'empty' RootVolume.Zones actually contains a single value of ""? I'll be honest: I don't fully understand the context here, but that seems like a potential source of error.